ice-adapter/src/main/java/com/faforever/iceadapter/debug/TelemetryDebugger.java

```diff
@@ -108,6 +107,8 @@ private void sendingLoop() {
 
                 log.trace("Sending telemetry message: {}", json);
                 websocketClient.send(json);
+            } catch (InterruptedException e) {
+                log.info("Sending loop interrupted");
```

Comment:
Catching the interrupt without re-interrupting while clear out the .isInterrupted() state and block the while loop from ending

https://chatgpt.com/share/67502077-b55c-800e-99d3-f6ede4960126

Reply:
You would be right if the code had `while (true)`.
But we have the use of ' while (!Thread.currentThread().isInterrupted())`. After that, the `while` ends and the thread stop doing work.

Reply:
If you catch the InterruptedException, `!Thread.isInterrupted()` returns always true.

Reply:
Yes, you're right, I forgot that this mechanism works like that. I'll fix it in the code now.